src/NHibernate/Hql/Ast/ANTLR/HqlSqlWalker.cs

```diff
@@ -749,7 +759,10 @@ IASTNode CreateFromFilterElement(IASTNode filterEntity, IASTNode alias)
 			{
 				log.Debug("createFromFilterElement() : processed filter FROM element.");
 			}
-	
+
+			if (fromElementFound)
+				return (IASTNode) adaptor.Nil();
```

Comment:
As I understand after quick look-around. There are some differences in the AST between HQL and LINQ. It seems that LINQ AST already contains `fromElement` in the tree, and HQL does not. 

We still need to do all these manipulations with the `fromElement`, but we do not want to add it to the tree (because it is already there).